src/auxiliary/special_elixirs.jl

```diff
@@ -290,7 +290,7 @@ function extract_initial_resolution(elixir, kwargs)
             return initial_refinement_level
         end
     catch e
-        if isa(e, UndefVarError)
+        if isa(e, ArgumentError)
             # get cells_per_dimension from the elixir
```

Comment:
Do you know why this change is necessary? And could you maybe add a comment explaining it?

Reply:
In `find_assignment`, we're doing
```julia
local result

walkexpr(expr) do x
    if something
        result = something_else
    end
end

return result
```
So this causes an `UndefVarError` when we tried to return `result`. Now, it will fail explicitly with an `ArgumentError`, so that we have a meaningful error message. Note that this also solves the problem where people ask in Slack why they are getting a `UndefVarError` when trying to do a convergence test with some elixir that doesn't have the assignment `initial_refinement_level`. Now, they will get `ArgumentError: assignment `initial_refinement_level` not found in expression`, which is at least better than before.

Reply:
I added a short comment explaining this and a more specific error, since I've seen multiple people asking about this specific problem in Slack.